test/support/utils_invoke.h

```diff
@@ -37,6 +37,52 @@ struct unique_kernel_name;
 template <typename Policy, int idx>
 using new_kernel_name = unique_kernel_name<typename ::std::decay<Policy>::type, idx>;
 
+constexpr bool
+use_explicit_kernel_names()
+{
+#ifndef TEST_EXPLICIT_KERNEL_NAMES
+    static_assert(false, "The state of TEST_EXPLICIT_KERNEL_NAMES is not defined");
```

Comment:
In case such preprocessor logic, in particular, some macro is not defined - usually a diagnostic is also on preprocessor level, like this:
https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/pstl/parallel_backend.h#L53

In general, we dont need such function `use_explicit_kernel_names()` at all...
It is simpler (and more clear) to use macro `TEST_EXPLICIT_KERNEL_NAMES` directly in the places of using (lines  67 and 79)